packages/grafana-data/src/field/fieldState.ts

```diff
@@ -38,9 +38,10 @@ export function getFrameDisplayName(frame: DataFrame, index?: number) {
 
 export function getFieldDisplayName(field: Field, frame?: DataFrame, allFrames?: DataFrame[]): string {
   const existingTitle = field.state?.displayName;
-  const calculatedForMultipleFrames = Boolean(field.state?.multipleFrames) && !Boolean(allFrames);
+  const canUseCachedDisplayName =
```

Comment:
I've modified the expression so it is no longer inverted, keeps tests working and respects the desired functionality.